### PR TITLE
Pull Rating Algorithm into Main

### PR DESCRIPTION
**Pull request by bpembert**
Also added some changes to the rating algorithm. Includes node modules we will want to delete.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
 const { Storage } = require('@google-cloud/storage');
+const { constants } = require('fs/promises');
+const { zip } = require('compressing');
+const { OutgoingMessage } = require('http');
+const { ChildProcess } = require('child_process');
 console.log(process.env.BUCKET_CREDENTIALS)
 if (process.env.BUCKET_CREDENTIALS == undefined) {
   console.log("Getting BUCKET_CREDENTIALS")
-  require('dotenv').config({path:__dirname+'/./../../../.env'})
+  require('dotenv').config({path:__dirname+'/./../../../../../.env'})
```
Comment (hugoday) [inline index.js:21]: This shouldn't be changed back

Comment (bpembert) [reply index.js]: Updated.

File: index.js [hunk 2]
```diff
@@ -1,14 +1,25 @@
 var express=require('express');
 var bodyParser = require('body-parser');
 const crypto = require('crypto')
+const cookieParser = require('cookie-parser');
```
Comment (hugoday) [inline index.js:4]:
```suggestion
```

File: index.js [hunk 3]
```diff
@@ -24,7 +35,9 @@ const upload = multer();
 var app=express();
 app.set("view engine", "ejs");
 app.use(bodyParser.urlencoded({ extended: true }));
-app.use(express.json({type:'application/json'}));
+app.use(cookieParser());
```
Comment (hugoday) [inline index.js:38]:
```suggestion
```

File: scripts/databaseSetup.js [hunk 4]
```diff
@@ -28,24 +28,25 @@ function setupDatabase() {
       CREATE TABLE IF NOT EXISTS packages (
         key INTEGER PRIMARY KEY,
         id TEXT NOT NULL,
-        name TEXT NOT NULL,
+        packageName TEXT NOT NULL,
```
Comment (hugoday) [inline scripts/databaseSetup.js:31]: This change will break a ton of stuff in index.js

File: index.js [hunk 5]
```diff
@@ -51,6 +69,9 @@ app.post('/packages', auth, (req, res) => {
   
   const packageQuery = req.body
   console.log("package query: " + JSON.stringify(packageQuery))
+  console.log("package query: " + JSON.stringify(packageQuery.map(pkg => pkg.Name)))
+  console.log(`SELECT * FROM packages WHERE name IN (${packageQuery.map(() => '?').join(', ')})`)
+
```
Comment (hugoday) [inline index.js:74]:
```suggestion
```

**Review by hugoday — CHANGES_REQUESTED** [review]
Also get rid of all the old views, styles, templates

**Pull request closed without merging**